### PR TITLE
Use a member function instead of a member variable to implement shape

### PR DESCRIPTION
**Pull request by serge-sans-paille**
This harmonizes ndarray, array and list to that respect.

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
travis is OK :-)

**Comment by pbrunet** [comment]
Yes but there are comments.

**Comment by serge-sans-paille** [comment]
updated!

File: pythran/pythonic/types/tuple.hpp [hunk 1]
```diff
@@ -296,11 +311,11 @@ namespace pythonic {
 
                 template<class... Types>
                     operator std::tuple<Types...>() const {
-                        return std::tuple<Types...>(to_tuple_impl<N>()(*this));
+                        return array_to_tuple(*this, typename utils::gens<N+1>::type{});
                     }
 
-                auto to_tuple() const -> decltype(to_tuple_impl<N>()(*this)) {
-                    return to_tuple_impl<N>()(*this);
```
Comment (pbrunet) [inline pythran/pythonic/types/tuple.hpp]: I like the new implementation but to_tuple_impl should be remove in this case. (It is not use anymore)

Comment (serge-sans-paille) [inline pythran/pythonic/types/tuple.hpp]: done

**Comment by serge-sans-paille** [comment]
@pbrunet ready to go!

**Pull request merged**